Library/Homebrew/formula.rb

```diff
@@ -1364,6 +1364,18 @@ def self.clear_installed_formulae_cache
     @installed = nil
   end
 
+  def self.comparison
```

Comment:
There's no location that jumps out to me as more appropriate. My first thought is to add formula_comparer.rb, but this still suffers from the "this is used by formulae and casks" issue and I'm hesitant to add another file to ./Library/Homebrew. Do you have a location in mind? Is there a common name for "formulae and casks"? Maybe "token_comparer.rb" or "full_token_comparer.rb"?